cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -169,6 +169,19 @@ inline compute::Expression UseBoringRefs(const compute::Expression& expr) {
   return compute::Expression{std::move(modified_call)};
 }
 
+Status CheckTable(std::shared_ptr<Buffer>& buf,
```

Comment:
Maybe `CheckPlanRuns` or `CheckPlanDoesNotFail`.  You could use `DeclarationToStatus` too.

Reply:
Turns out this function is no longer needed, so I removed it.